### PR TITLE
View more details on projects

### PR DESCRIPTION
**Pull request by ericawright**
opening for visibility

this will:
- allow user to inspect more details of each project
- rearrange some components
- create Modal and NewProjectForm components
- reuse modal for more details view
- allow editing a post

TODO:
- ~~submit edited content~~
- ~~rename and rearrange component files~~

### CODE REVIEW

**Comment by ericawright** [comment]
I think this is ready for review now

File: src/models.rs [hunk 1]
```diff
@@ -25,4 +25,18 @@ pub struct NewProject {
     pub name: Option<String>,
     pub link: String,
     pub email: String,
-}
\ No newline at end of file
+}
+
+#[derive(Queryable, Debug, Serialize, Deserialize, Insertable, AsChangeset)]
```
Comment (bwinton) [inline src/models.rs:30]: What if we derived `AsChangeset` on the `Project` struct instead?
Then you should be able to use it and remove the `EditProject` struct…

Comment (bwinton) [reply src/models.rs:30]: (From https://dzone.com/articles/creating-a-rest-api-in-rust-using-rocket-and-diese )

**Review by bwinton — APPROVED** [review]
Mostly looks good, but I think we could do something a little nicer with the database…

**Pull request merged**